module/data/actor/templates/attributes.mjs

```diff
@@ -145,6 +149,83 @@ export default class AttributesFields {
 
   /* -------------------------------------------- */
 
+  /**
+   * Calculate encumbrance details for an Actor.
+   * @this {CharacterData|NPCData|VehicleData}
+   * @param {object} rollData  The Actor's roll data.
+   */
+  static prepareEncumbrance(rollData) {
+    const config = CONFIG.DND5E.encumbrance;
+    const encumbrance = this.attributes.encumbrance ??= {};
+    const baseUnits = CONFIG.DND5E.encumbrance.baseUnits[this.parent.type]
+      ?? CONFIG.DND5E.encumbrance.baseUnits.default;
+    const unitSystem = game.settings.get("dnd5e", "metricWeightUnits") ? "metric" : "imperial";
+
+    // Get the total weight from items
+    let weight = this.parent.items
+      .filter(item => !item.container)
+      .reduce((weight, item) => weight + (item.system.totalWeightIn?.(baseUnits[unitSystem]) ?? 0), 0);
+
+    // [Optional] add Currency Weight (for non-transformed actors)
+    const currency = this.currency;
+    if ( game.settings.get("dnd5e", "currencyWeight") && currency ) {
+      const numCoins = Object.values(currency).reduce((val, denom) => val + Math.max(denom, 0), 0);
+      const currencyPerWeight = config.currencyPerWeight[unitSystem];
+      weight += numCoins / currencyPerWeight;
+    }
+
+    // Determine the Encumbrance size class
+    const keys = Object.keys(CONFIG.DND5E.actorSizes);
+    const index = keys.findIndex(k => k === this.traits.size);
+    const sizeConfig = CONFIG.DND5E.actorSizes[
+      keys[this.parent.flags.dnd5e?.powerfulBuild ? Math.min(index + 1, keys.length - 1) : index]
+    ];
+    const mod = sizeConfig?.capacityMultiplier ?? sizeConfig?.token ?? 1;
```

Comment:
I know this was existing code but calling this `sizeMod` might make things a bit clearer in the calculations.